common/io_lock.c

```diff
@@ -0,0 +1,74 @@
+#include "io_lock.h"
+#include <assert.h>
+#include <ccan/io/io_plan.h>
+
+struct io_lock {
+	bool free;
```

Comment:
nitpick mode: enabled!

I would `bool locked` this, as free has other connotations which are unwelcome here....